test/integration/block_low_follower_users_test.js

```diff
@@ -0,0 +1,86 @@
+var system = require('system');
+var mainUser = system.env.BT_TEST_MAIN_USER;
+var mainPass = system.env.BT_TEST_MAIN_PASS;
+var lowUser = system.env.BT_TEST_LOW_FOLLOWER_USER;
+var lowPass = system.env.BT_TEST_LOW_FOLLOWER_PASS;
+
+var host = 'http://localhost:3000';
+
+function checkBoxes() {
+  return [
+    document.querySelector('#block_new_accounts').checked,
+    document.querySelector('#block_low_followers').checked,
+    document.querySelector('#share_blocks').checked,
+    document.querySelector('#follow_blocktogether').checked
+  ];
+}
+
+casper.test.begin('Block low follower users', 2, function(test) {
+  casper.start(host, function() {
+    this.click('#block_low_followers');
+    return this.fill('form[action*="/auth/twitter"]', {}, true);
+  });
+
+  casper.waitForSelector('#oauth_form', function() {
+    return this.fill(
+      'form[id="oauth_form"]',
+      // NB: must use single quotes
+      { 'session[username_or_email]': mainUser, 'session[password]': mainPass }, true);
+  });
+
+  casper.waitForSelector('.container-fluid', function() {
```

Comment:
Ah, that makes sense- for instance if we got redirected to a Twitter error
page rather than back to the app.

On Sep 13, 2016 17:38, "Alex Dunn" <notifications@github.com> wrote:

> In test/integration/block_low_follower_users_test.js
> <https://github.com/jsha/blocktogether/pull/231#discussion_r78670185>:
>
> > +}
> > +
> > +casper.test.begin('Block low follower users', 2, function(test) {
> > +  casper.start(host, function() {
> > +    this.click('#block_low_followers');
> > +    return this.fill('form[action*="/auth/twitter"]', {}, true);
> > +  });
> > +
> > +  casper.waitForSelector('#oauth_form', function() {
> > +    return this.fill(
> > +      'form[id="oauth_form"]',
> > +      // NB: must use single quotes
> > +      { 'session[username_or_email]': mainUser, 'session[password]': mainPass }, true);
> > +  });
> > +
> > +  casper.waitForSelector('.container-fluid', function() {
>
> Just to be doubly sure it's the right page; I'm not sure if it's possible
> that Casper would see the body of the Twitter login page and think that's
> the selector it's looking for.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsha/blocktogether/pull/231/files/fb43cb3e53eda67f2d2876d4b507452243f6a854#r78670185>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANcLXH6kL0DfAIeI2DFtN4lu7IkBZPBks5qp0IUgaJpZM4J6IVj>
> .
>